### DISCUSSION TITLE
Blank screen on first boot

### DISCUSSION THREAD

**Discussion by Shinyrr** [sanni/cartreader#686] [Need help]
I recently started a build of the v5 cart reader, I've gone through the entirety of the process and double and triple checked steps along the way. My issue is that the initial test of the power on and checking gameboy functionality is not working, I get power, the device seems to turn on and theres a green light on the mega board, however the display doesnt seem to be functional beyond turning on. I've double checked all the solder points and theres no bridges, I've reflashed the mega twice, formatted the SD card multiple times and I'm just at a loss as to what could be causing this

![IMG_2644](https://user-images.githubusercontent.com/122844045/212791086-796e962d-0dee-4fb3-85ab-5eb70302645b.jpg)

**Comment by sanni** [comment]
Most likely: Try flashing the code with more modules disabled by adding // in front of systems you don't need: https://github.com/sanni/cartreader/blob/master/Cart_Reader/Cart_Reader.ino#L78

Small chance: Check that the display cable is inserted the right way on both the Main PCB and the LCD.  For example it could be that the IDC connector on the LCD was put on by the factory the wrong way. The notch should be at the top like in this picture: https://github.com/sanni/cartreader/wiki/How-to-build#6-the-first-test

**Reply by Shinyrr** [reply]
so I tested both of those theories and neither seems to resolve it, I went from the 10.2 files to the 12 beta and limited it down to just the GB slot and it still happens, I also confirmed the cables are correct. I went over the board with a multimeter and think I found the issue however, I'm getting continuity between pins 6 and 9 on the EXP1 and EXP2 ports on the PCB which as far as I can tell should not be a thing based on the diagrams on the build guide, so I think the PCB might be bad but can you confirm if this is correct or not?

**Reply by Cephiros** [reply]
Can you Try booting at 5v without a game(if using gba) and make sure 3.3 isnt causing the problem.

**Reply by Shinyrr** [reply]
Yes, this occurs on both 5v and 3.3v and it happens with no game inserted

**Reply by Cephiros** [reply]
Could you have too many modules enabled? 
`// add/remove // before #define to disable/enable modules you
// don't need/need to save program storage space and dynamic memory
// If you only get an empty or "Press Button" screen after flashing
// you have enabled too many modules`

**Reply by Shinyrr** [reply]
as mentioned above, I did that already and even went as far as to limit it to just the GB module to rule that issue out

**Comment by sanni** [comment]
What you can try is change Cart_Reader.ino like this:

````
//******************************************
// !!! CHOOSE HARDWARE VERSION !!!
//******************************************
// Remove // in front of the line with your hardware version
// #define HW5
// #define HW4
// #define HW3
// #define HW2
// #define HW1
#define SERIAL_MONITOR
````

Upload and then open the Arduino Serial Monitor. It's found under Tools -> Serial Monitor in the Arduino IDE, needed settings: 9600 baud, no line ending.

More info about the Serial Monitor: https://github.com/sanni/cartreader/wiki/Serial-Monitor

Then see if you get an SD Error too, the display and the SD card share some pins, so if both don't work we could limit the issue to those pins.

````
Cartridge Reader
2023 github.com/sanni
SD Error

Press Button...
````


> I'm getting continuity between pins 6 and 9 on the EXP1 and EXP2 ports on the PCB

 I get 10 megaOhm resistance between D51 to GND. And 28 megaOhm resistance between D13 and GND. No direct continuity.
Assuming you used the pin numbers as shown in this image:

![image](https://dl.dropboxusercontent.com/s/tj0xd3n2lvterdr/display_connect_1.jpg?dl=1)     

In my schematic I used a different numbering of the pins

![image](https://raw.githubusercontent.com/sanni/cartreader/master/hardware/main_pcb_hw5/main_pcb_hw5_schematic.jpg)

**Reply by Shinyrr** [reply]
Sorry for the late reply, I finally was able to test using the serial monitor and I am in fact getting an SD error

**Reply by sanni** [reply]
That does make it easier since the SD card only used 4 pins. Arduino D50, D51, D52 and D53. They are all found on the small 6x2 header. 
Can you re-solder all the pins both on the Cart Reader PCB and the Mega Pro to make sure they have good contact?

![qffffaffffff](https://user-images.githubusercontent.com/19169494/213994470-7e05a786-7bce-4679-9cfd-23cb053175c3.jpeg)

![rqwrqqfqrrqrq](https://user-images.githubusercontent.com/19169494/213994476-73da72a9-85a7-4aac-95f7-edccae2f59f5.jpeg)

And also check the Atmega itself for solder bridges on the SD pins from the factory.

![fqwraqfqffqfqfqfq](https://user-images.githubusercontent.com/19169494/213995541-164888e7-0a33-4cd4-9c6a-a37a40cffd48.jpg)

**Reply by Shinyrr** [reply]
I went over everything again and resoldered those points as well as checked for bridges on the chip and nothing, I did however notice something on the board I didnt see before, but it looks like there is an area just under where one of the display connectors goes that appears to be damaged in some way you can see the picture attached. I can see traces exposed so I'm wondering if this is damaged and are connected to those pins you mentioned.
![IMG_2676](https://user-images.githubusercontent.com/122844045/215299779-01591d69-aa8f-431a-8155-b71edd5374cb.jpg)

**Reply by sanni** [reply]
Those traces are just for the rotary encoder button and status LED.
